lib/Core/Site/Pagination/Pagerfanta/ContentSearchHitAdapter.php

```diff
@@ -40,14 +40,13 @@ public function __construct(Query $query, FindService $findService)
      */
     public function getNbResults()
     {
-        if (isset($this->nbResults)) {
-            return $this->nbResults;
+        if ($this->isSearchResultNotCached()) {
+            $countQuery = clone $this->query;
```

Comment:
c

Reply:
Yes, otherwise injected query would be modified by `getNbResults()` and `getSlice()` methods.